app/controllers/assignments_controller.rb

```diff
@@ -262,6 +264,13 @@ def set_case
     @case_transitions = @case.transitions.decorate
   end
 
+  def check_case_not_deleted
+    if @case.nil? && Case::Base.unscoped.find(params[:case_id])
+      flash[:notice] = 'Case has been deleted.'
+      redirect_to open_filter_path
+    end
+  end
+
```

Comment:
Recommend something like this:

```
  def redirect_on_nil_case!
    return if Case::Base.exists?(params[:case_id])
    redirect_to open_filter_path, alert: 'Case has been deleted.'
  end
```

Reply:
As discussed, I'd like to keep the logic as-is so that it specifically relates to deleted cases. Renamed method more or less as you propose - now `redirect_on_deleted_case!`